src/bridge_node.cpp

```diff
@@ -44,27 +44,27 @@ int main(int argc, char** argv)
 
     ros::param::param<std::string>("~server_addr", tcp_addr, "127.0.0.1");
     ros::param::param<std::string>("~server_port", tcp_port, "6002");
-    ros::param::param<std::string>("~sys_name", sys_name, "sam-auv-1");
+    ros::param::param<std::string>("~system_name", sys_name, "imc_ros_bridge");
     ros::param::param<int>("~imc_id", imc_id, 30);
 
     IMCHandle imc_handle(tcp_addr, tcp_port, sys_name, imc_id);
 
-    ros_to_imc::BridgeServer<std_msgs::Empty, IMC::Heartbeat> heartbeat_server(ros_node, imc_handle, "/heartbeat");
-    ros_to_imc::BridgeServer<sensor_msgs::NavSatFix, IMC::GpsFix> gpsfix_server(ros_node, imc_handle, "/gps_fix");
-    ros_to_imc::BridgeServer<geometry_msgs::Pose, IMC::Goto> goto_server_dummy(ros_node, imc_handle, "/goto_input");
-    ros_to_imc::BridgeServer<sensor_msgs::NavSatFix, IMC::GpsNavData> gpsnavdata_server(ros_node, imc_handle, "/gps_nav_data");
-    ros_to_imc::BridgeServer<sensor_msgs::NavSatFix, IMC::EstimatedState> estimatedstate_server(ros_node, imc_handle, "/estimated_state");
-    ros_to_imc::BridgeServer<imc_ros_bridge::PlanControlState, IMC::PlanControlState> plancontrolstate_server(ros_node, imc_handle, "/plan_control_state");
+    ros_to_imc::BridgeServer<std_msgs::Empty, IMC::Heartbeat> heartbeat_server(ros_node, imc_handle, sys_name+"/heartbeat");
```

Comment:
Pro tip: if you just emit the first "/" in the topic name, it will get the name "/node_name/topic_name".

Reply:
I am trying to get multiple bridges to run at the same time, so being able to put the imc stuff under specific vehicles seemed like a good start :)